src/main/kotlin/com/infusionvlc/somniumserver/dreams/persistence/DreamRepository.kt

```diff
@@ -1,5 +1,10 @@
 package com.infusionvlc.somniumserver.dreams.persistence
 
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.Pageable
 import org.springframework.data.repository.PagingAndSortingRepository
 
-interface DreamRepository : PagingAndSortingRepository<DreamEntity, Long>
+interface DreamRepository : PagingAndSortingRepository<DreamEntity, Long> {
+
+  fun findByTitleContainingIgnoreCase(title: String, pageable: Pageable): Page<DreamEntity>
```

Comment:
I think it's much more readable than JPQL [https://docs.spring.io/spring-data/jpa/docs/1.5.1.RELEASE/reference/html/jpa.repositories.html#jpa.query-methods.query-creation](documentation here), I recommend you use it whenever you can.

It is the same as using JDBC, it could fail if the DB has too many connections or something like that.

Reply:
In theory that is managed automagically. As I said, the only exceptions I've seen using JPA and spring-data are caused because your entities are bad structured which lead to bad database schemas.

What I'm thinking about is to make an additional layer between this classes and usecases because as you noticed for sure we are mapping through domains at the usecases and this is not 100% CLEAN. Besides, the spring team chose bad names for these classes (`CrudRepository`, `PagingAndSortingRepository`) because in my opinion these are DAOs 🤔 . 

Maybe we could create an intermediate class which receive these classes as datasources and perform necessary mappings. We can transform `java.util.Optionals` to arrow's `Option` and use `Try` in there if we desire 👍 .